src/stores/connectors/eth1/eth1.go

```diff
@@ -1,299 +1,46 @@
-	return result, nil
-}
-
-func (c Connector) SignTransaction(ctx context.Context, addr string, chainID *big.Int, tx *types.Transaction) ([]byte, error) {
-	logger := c.logger.With("address", addr)
-
-	result, err := c.store.SignTransaction(ctx, addr, chainID, tx)
-	if err != nil {
-		return nil, err
-	}
-
-	logger.Debug("transaction signed successfully")
-	return result, nil
+	store  stores.KeyStore
+	logger log.Logger
+	db     database.ETH1Accounts
 }
 
-func (c Connector) SignEEA(ctx context.Context, addr string, chainID *big.Int, tx *types.Transaction, args *ethereum.PrivateArgs) ([]byte, error) {
-	logger := c.logger.With("address", addr)
+var _ stores.Eth1Store = Connector{}
 
-	result, err := c.store.SignEEA(ctx, addr, chainID, tx, args)
-	if err != nil {
-		return nil, err
-	}
-
-	logger.Debug("EEA transaction signed successfully")
-	return result, nil
-}
-
-func (c Connector) SignPrivate(ctx context.Context, addr string, tx *quorumtypes.Transaction) ([]byte, error) {
-	logger := c.logger.With("address", addr)
-
-	result, err := c.store.SignPrivate(ctx, addr, tx)
-	if err != nil {
-		return nil, err
-	}
-
-	logger.Debug("private transaction signed successfully")
-	return result, nil
+var eth1Algo = &entities.Algorithm{
+	Type:          entities.Ecdsa,
+	EllipticCurve: entities.Secp256k1,
 }
 
-func (c Connector) ECRevocer(ctx context.Context, data, sig []byte) (string, error) {
-	result, err := c.store.ECRevocer(ctx, data, sig)
-	if err != nil {
-		return "", err
-	}
-
-	c.logger.Debug("ethereum account recovered successfully from signature")
-	return result, nil
-}
-
-func (c Connector) Verify(ctx context.Context, addr string, data, sig []byte) error {
-	err := c.store.Verify(ctx, addr, data, sig)
-	if err != nil {
-		return err
-	}
-
-	c.logger.Debug("data verified successfully")
-	return nil
-}
-
-func (c Connector) VerifyTypedData(ctx context.Context, addr string, typedData *core.TypedData, sig []byte) error {
-	err := c.store.VerifyTypedData(ctx, addr, typedData, sig)
-	if err != nil {
-		return err
-	}
-
-	c.logger.Debug("typed data verified successfully")
-	return nil
-}
-
-func (c Connector) Encrypt(ctx context.Context, addr string, data []byte) ([]byte, error) {
-	logger := c.logger.With("address", addr)
-
-	result, err := c.store.Encrypt(ctx, addr, data)
-	if err != nil {
-		return nil, err
-	}
-
-	logger.Info("data encrypted successfully")
-	return result, nil
-}
-
-func (c Connector) Decrypt(ctx context.Context, addr string, data []byte) ([]byte, error) {
-	logger := c.logger.With("address", addr)
-
-	result, err := c.store.Decrypt(ctx, addr, data)
-	if err != nil {
-		return nil, err
+func NewEth1Connector(store stores.KeyStore, db database.ETH1Accounts, logger log.Logger) *Connector {
+	return &Connector{
+		store:  store,
+		logger: logger,
+		db:     db,
+	}
+}
+
+func parseKey(key *entities.Key, attr *entities.Attributes) *entities.ETH1Account {
+	pubKey, _ := crypto.UnmarshalPubkey(key.PublicKey)
```

Comment:
This is being a long discussion with @darioAnongba . The conclusion is we ignore errors in case we control the data flow and we can guarantee no error can occur. In this case the PublicKey is something we already validated on the infra layer so that at the time it reaches this part of the code we can ignore the error checks